pulseguardian/guardian.py

```diff
@@ -219,22 +228,62 @@ def back_to_normal_email(self, user, queue_data):
            queue_data['messages'], self.del_queue_size)
 
         if self.emails and user.email is not None:
-            sendemail(subject=subject, from_addr=config.email_from,
-                      to_addrs=[user.email], username=config.email_account,
-                      password=config.email_password, text_data=body,
-                      server=config.email_smtp_server,
-                      port=config.email_smtp_port,
-                      use_ssl=config.email_ssl)
+            self._sendemail(subject=subject, to_addrs=[user.email], text_data=body)
+
+    def notify_connection_error(self, e):
+        if self.emails and not self._connection_error_notified:
+            admin_emails = [user.email for user in User.query.filter_by(admin=True)]
+            subject = "Pulse error: Can't connect to Pulse"
+            body = '''Could not connect to Pulse.
+Details:
+Rabbit Url: {0}
+Rabbit User: {1}
+Error:
+{2}
+'''.format(config.rabbit_management_url, config.rabbit_user, e)
+
+            self._sendemail(subject=subject, to_addrs=admin_emails, text_data=body)
+            logging.error(subject + body)
+            self._connection_error_notified = True
+
+    def notify_unknown_error(self, e):
+        if self.emails and not self._unknown_error_notified:
+            admin_emails = [user.email for user in User.query.filter_by(admin=True)]
+            subject = "Pulse error: Unknown error"
+            body = '''Unknown error occured.
+Details:
+{0}
+'''.format(e)
+            self._sendemail(subject=subject, to_addrs=admin_emails, text_data=body)
+            logging.error(subject + body)
+            self._unknown_error_notified = True
 
     def guard(self):
         logging.info("PulseGuardian started")
-        while True:
-            queues = self.api.queues()
-            if queues:
-                self.monitor_queues(queues)
-            self.clear_deleted_queues(queues)
-            time.sleep(config.polling_interval)
 
+        while True:
+            try:
+                queues = self.api.queues()
+                if queues:
+                    self.monitor_queues(queues)
+                self.clear_deleted_queues(queues)
+
+                if self._connection_error_notified or self._unknown_error_notified:
+                    self._reset_notification_error_params()
+            except TypeError as te:
```

Comment:
Having now tested this a bit, what would be really great would be to include the traceback in the email as well. You'll have to use the sys.exc_info() function to extract it, and then the traceback module to format it.